x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalVerifier.java

```diff
@@ -56,7 +59,8 @@ void checkPlanConsistency(PhysicalPlan optimizedPlan, Failures failures, Failure
                 }
             }
 
-            if (p instanceof ExecutesOn ex && ex.executesOn() == ExecutesOn.ExecuteLocation.REMOTE) {
+            // This check applies only for general physical plans (isLocal == false)
+            if (isLocal == false && p instanceof ExecutesOn ex && ex.executesOn() == ExecutesOn.ExecuteLocation.REMOTE) {
```

Comment:
We could also add a check for the converse - after physical optimization on the remote, there shouldn't be any coordinator-only nodes in the plan.